### PR TITLE
Login

### PR DESCRIPTION
**Pull request by JohnOberhauser**
I will have a part 2 for login, but I'm going to wait for the module changes to be finished before I start touching use-cases

- Updating login screen to reflect designs (portrait and landscape)
- Creating select server screen
- Adding navigation for login flow
- Adding some window size class functions

.

.

### CODE REVIEW

File: app/src/main/java/org/mozilla/social/MainViewModel.kt [hunk 1]
```diff
@@ -26,7 +27,7 @@ class MainViewModel(
         viewModelScope.launch(Dispatchers.Main) {
             isSignedInFlow().collectLatest {
                 if (!it) {
-                    navigateTo(NavigationDestination.Login)
+                    navigateTo(AuthNavigationDestination.Login)
```
Comment (devotaaabel) [inline app/src/main/java/org/mozilla/social/MainViewModel.kt:30]: Shouldn't the old value here be correct? That's the destination of the nested graph in the main graph

File: core/designsystem/src/main/res/font/zillaslab_bold.ttf
Comment (devotaaabel) [inline core/designsystem/src/main/res/font/zillaslab_bold.ttf:1]: Since when are we using custom fonts?

**Review by devotaaabel — COMMENTED** [review]
I'll wait until opening another PR until this merges, please don't merge this until the other two are merged. I think it'll be a lot easier to resolve conflicts from your side if they arise

File: core/designsystem/src/main/res/font/zillaslab_bold.ttf
Comment (JohnOberhauser) [reply core/designsystem/src/main/res/font/zillaslab_bold.ttf:1]: This is just for the `mozilla.social` text on the login screen

File: app/src/main/java/org/mozilla/social/MainViewModel.kt (hunk 1 above)
Comment (JohnOberhauser) [reply app/src/main/java/org/mozilla/social/MainViewModel.kt]: oops, yeah

File: feature/auth/src/main/java/org/mozilla/social/feature/auth/chooseServer/ChooseServerViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import org.mozilla.social.core.usecase.mastodon.auth.Login
+
+class ChooseServerViewModel(
+    private val login: Login,
+) : ViewModel(), ChooseServerInteractions {
+
+    private val _uiState = MutableStateFlow(ChooseServerUiState())
+    val uiState = _uiState.asStateFlow()
+
+    @Suppress("SwallowedException")
```
Comment (devotaaabel) [inline feature/auth/src/main/java/org/mozilla/social/feature/auth/chooseServer/ChooseServerViewModel.kt:19]: Is this supposed to be here?

File: feature/auth/src/main/java/org/mozilla/social/feature/auth/login/LoginScreen.kt [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+import org.mozilla.social.core.designsystem.theme.MoSoSpacing
+import org.mozilla.social.core.designsystem.theme.MoSoTheme
+import org.mozilla.social.feature.auth.R
+
+@Composable
+internal fun LoginScreen(
+    viewModel: LoginViewModel = koinViewModel(),
+) {
+    val configuration = LocalConfiguration.current
+    if (configuration.orientation == Configuration.ORIENTATION_LANDSCAPE) {
```
Comment (devotaaabel) [inline feature/auth/src/main/java/org/mozilla/social/feature/auth/login/LoginScreen.kt:49]: Can you use the compact size class here that was in the old login screen plz? That's supposed to be more flexible than `ORIENTATION_LANDSCAPE`

Comment (JohnOberhauser) [reply feature/auth/src/main/java/org/mozilla/social/feature/auth/login/LoginScreen.kt:49]: I added a commit to create some window size class convenience functions.  I was going to switch to using window class as per your recommendation, but as I was deciding which window size classes should show which layout, I realized that using orientation is the right way to go in this case, based on the designs.  There isn't a clear combination of height / width window size classes that fit well here, it's really just portrait vs landscape

File: feature/auth/src/main/java/org/mozilla/social/feature/auth/chooseServer/ChooseServerViewModel.kt (hunk 2 above)
Comment (JohnOberhauser) [reply feature/auth/src/main/java/org/mozilla/social/feature/auth/chooseServer/ChooseServerViewModel.kt]: nope, removed

File: feature/auth/src/main/java/org/mozilla/social/feature/auth/login/LoginScreen.kt (hunk 3 above)
Comment (devotaaabel) [reply feature/auth/src/main/java/org/mozilla/social/feature/auth/login/LoginScreen.kt:49]: IMO @JohnOberhauser what you could do here is something like:
if compact height and width - only show login section
if compact width and medium/extended height - show image and login section vertically
if compact height and medium/extended width - show image and login section horizontally

https://developer.android.com/guide/topics/large-screens/migrate-to-responsive-layouts#things-to-avoid

**Review by devotaaabel — APPROVED** [review]
Please still address the last feedback in a follow up

File: feature/auth/src/main/java/org/mozilla/social/feature/auth/login/LoginScreen.kt (hunk 3 above)
Comment (JohnOberhauser) [reply feature/auth/src/main/java/org/mozilla/social/feature/auth/login/LoginScreen.kt:49]: Uhg, it's annoying that multi-window mode can cause the app to be in landscape when height is greater than width

**Pull request merged**